src/server/middleware/repoproxy/proxy.go

```diff
@@ -94,12 +94,22 @@ func handleManifest(w http.ResponseWriter, r *http.Request, next http.Handler) e
 	if err != nil {
 		return err
 	}
-	if !canProxy(p) || proxyCtl.UseLocalManifest(ctx, art) {
+	remote, err := proxy.NewRemoteHelper(p.RegistryID)
+	if err != nil {
+		log.Errorf("failed to get remote registry, registry id: %v", p.RegistryID)
+		next.ServeHTTP(w, r)
+		return nil
+	}
+	useLocal, err := proxyCtl.UseLocalManifest(ctx, art, remote)
+	if err != nil {
+		return err
+	}
+	if !canProxy(p) || useLocal {
```

Comment:
Put the `proxyCtl.UseLocalManifest(ctx, art, remote)` after the checking of `canProxy(p)` to avoid useless checking?

Reply:
yes, updated